### PR TITLE
Custom fields from request in audit logger

### PR DESCRIPTION
**Pull request by sfaizanh**
Currently there is not way to log custom fields in auditLogger, This PR support custom logging for auditlogger
```
restify.plugins.auditLogger({
    log: bunyan.createLogger({
    name: 'audit',
    streams: [
        {
            level: 'info',
            type: 'raw'
        }
    ]
    }),
    event: 'after',
    customFields: ['user']
})
```
Pre-Submission Checklist
- [x] Opened an issue discussing these changes before opening the PR
- [x] Ran the linter and tests via make prepush
- [x] Included comprehensive and convincing tests for changes

**Closes:**
Issue #1343

### CODE REVIEW

**Comment by sfaizanh** [comment]
Node stable build on Travis failed (but all build are passed). Looks like a random failure with npm. Closing (will reopen to retrigger Travis build).

**Comment by retrohacker** [comment]
If you agree with the changes, feel free to merge @hekike!

File: lib/plugins/audit.js [hunk 1]
```diff
@@ -50,6 +51,7 @@ function getResponseHeaders(res) {
  * the audit log object programmatically
  * @param {boolean} [opts.printLog=true] - Whether to print the log
  * via the logger.
+ * @param {Array<String>} [opts.customFields=[]] - Custom fields for logger
  * @returns {Function} Handler
  * @fires audit when an audit log has been generated
  * @example
```
Comment (hekike) [inline lib/plugins/audit.js]: could you please provide a JSDoc example?

**Comment by hekike** [comment]
@sfaizanh @retrohacker small comment about docs.

File: lib/plugins/audit.js (hunk 1 above)
Comment (sfaizanh) [reply lib/plugins/audit.js]: Should i change `Array<String>` to `string[]`?

Comment (hekike) [reply lib/plugins/audit.js]: Yes, please. Also you can add a new example to JSDoc like:
```
* @example
* <caption>Some text</caption>
* Some JS Code

**Comment by sfaizanh** [comment]
Hi @hekike, any plan to merge?

File: lib/plugins/audit.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ var assert = require('assert-plus');
 var bunyan = require('bunyan');
 var HttpError = require('restify-errors').HttpError;
 var VError = require('verror');
+var lodash = require('lodash');
```
Comment (hekike) [inline lib/plugins/audit.js]: could you change this line to `var _ = require('lodash');`?
We call lodash `_` in other files in restidf.

**Comment by hekike** [comment]
@sfaizanh yes, sry for the delay. One last comment.

**Comment by sfaizanh** [comment]
I haven't changed https://github.com/sfaizanh/node-restify/blob/6f6ae16c2f013f6e35b3b50147d99fcbdd176f54/test/plugins/audit.test.js#L89 but `lodash` as variable is used, should i change it?

**Comment by DonutEspresso** [comment]
This seems overlapping with the existing `context` option:
https://github.com/restify/node-restify/blob/master/test/plugins/audit.test.js#L625

The `context` option allows specifying fields as well as ability to format. Do we need both of these options?

**Comment by retrohacker** [comment]
I wasn't aware of the `context` key, though it is documented: http://restify.com/docs/plugins-api/#auditlogger

Does this do what you were needing it to @sfaizanh?  Sorry that we got so far with this PR before this was brought up.

**Comment by hekike** [comment]
I'd prefer to move forward with this PR either way.
@sfaizanh did you have time to check out the `context` key? Would it solve your use-case?

**Comment by sfaizanh** [comment]
Ok I’ll test this

**Pull request closed without merging**